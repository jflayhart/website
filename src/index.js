import React from 'react'
import ReactDOM from 'react-dom'
import ThemeProvider from './components/ThemeProvider'
import Placeholder from './components/Placeholder'

export const App = () => (
  <ThemeProvider>
    <Placeholder />
  </ThemeProvider>
)

ReactDOM.render(<App />, document.getElementById('root'))
