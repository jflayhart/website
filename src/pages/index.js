import React from 'react'
import ReactDOM from 'react-dom'
import { createGlobalStyle } from 'styled-components'
import Main from '../components/Main'

const GlobalStyle = createGlobalStyle`
  body {
    height: 100%;
    margin: 0;
    font-family: ${props => props.theme.fontFamily};
    color: ${props => props.theme.fontColor};
    background: #FFFFF8;
  }
  .stripe-one,
  .stripe-two,
  .stripe-three,
  .stripe-four {
    position: absolute;
    left: -25%;
    height: 20px;
    width: 100%;
    transform: rotate(-45deg) skew(-45deg);
  }
  .stripe-one {
    background-color: #a0d2c8;
  }
  .stripe-two {
    background-color: #a1d29e;
    top: 30px;
  }
  .stripe-three {
    background-color: #f16146;
    top: 60px;
  }
  .stripe-four {
    background-color: #eb1f44;
    top: 90px;
  }
`

export const Home = () => (
  <>
    <GlobalStyle />
    <div className="stripe-one" />
    <div className="stripe-two" />
    <div className="stripe-three" />
    <div className="stripe-four" />
    <Main />
  </>
)

export default Home
