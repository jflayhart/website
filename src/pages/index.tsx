import { createGlobalStyle } from 'styled-components'
import Main from '../components/Main'

const GlobalStyle = createGlobalStyle`
  body {
    height: 100%;
    margin: 0;
    font-family: ${(props: any) => props.theme.fontFamily};
    color: ${(props: any) => props.theme.fontColor};
    background: #FFFFF8;
  }
  .stripe-one,
  .stripe-two,
  .stripe-three,
  .stripe-four {
    position: absolute;
    left: -25%;
    height: 20px;
    width: 100%;
    transform: rotate(-45deg) skew(-45deg);
  }
  .stripe-one {
    background-color: #a0d2c8;
  }
  .stripe-two {
    background-color: #a1d29e;
    top: 30px;
  }
  .stripe-three {
    background-color: #f16146;
    top: 60px;
  }
  .stripe-four {
    background-color: #eb1f44;
    top: 90px;
  }
`

export function getStaticProps() {
  return { props: {} }
}

const Home = () => (
  <>
    <GlobalStyle />
    <div className="stripe-one" />
    <div className="stripe-two" />
    <div className="stripe-three" />
    <div className="stripe-four" />
    <Main />
  </>
)

export default Home
