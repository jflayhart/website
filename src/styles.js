import { createMuiTheme } from 'material-ui/styles'
import blue from 'material-ui/colors/blue'
import green from 'material-ui/colors/green'
import red from 'material-ui/colors/red'

// https://material-ui-next.com/customization/themes
export default createMuiTheme({
  palette: {
    type: 'light',
    primary: blue,
    secondary: { main: green[700] },
    error: red,
  },
  typography: {
    fontFamily: 'Orbitron, sans-serif',
  }
})
