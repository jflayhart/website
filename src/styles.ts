export default {
  fontFamily: 'Helvetica, sans-serif',
  fontColor: '#333',
  accentColor: '#a1d29e'
}
