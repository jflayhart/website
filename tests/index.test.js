import React from 'react'
import renderer from 'react-test-renderer';
import { App } from '../src/index'

jest.mock('react-dom')

test('should render valid JSX (snapshot)', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
