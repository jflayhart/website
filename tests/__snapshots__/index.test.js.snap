// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render valid JSX (snapshot) 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <img
    src="test-file-stub"
    style={
      Object {
        "animation": "",
      }
    }
    width="200"
  />
  <div>
    Hmm... This site hasn't 
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "green",
          "cursor": "pointer",
          "display": "inline-block",
          "marginBottom": 20,
        }
      }
    >
      hatched
    </span>
     quite yet
  </div>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 10px",
        }
      }
    >
      <a
        href="https://www.facebook.com/jjflayhart"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          src="test-file-stub"
          width="40"
        />
      </a>
    </div>
    <div
      style={
        Object {
          "margin": "0 10px",
        }
      }
    >
      <a
        href="https://twitter.com/joshflayhart"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          src="test-file-stub"
          width="40"
        />
      </a>
    </div>
    <div
      style={
        Object {
          "margin": "0 10px",
        }
      }
    >
      <a
        href="https://www.linkedin.com/in/jflayhart"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          src="test-file-stub"
          width="40"
        />
      </a>
    </div>
    <div
      style={
        Object {
          "margin": "0 10px",
        }
      }
    >
      <a
        href="https://github.com/jflayhart"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          src="test-file-stub"
          width="40"
        />
      </a>
    </div>
  </div>
</div>
`;
